assets/javascripts/discourse/components/solved-unaccept-answer-button.gjs

```diff
@@ -41,23 +43,52 @@ function unacceptPost(post) {
 
 export default class SolvedUnacceptAnswerButton extends Component {
   @service appEvents;
+  @service siteSettings;
 
   @action
   unacceptAnswer() {
     unacceptAnswer(this.args.post, this.appEvents);
   }
 
+  get solvedBy() {
+    const username = this.args.post.topic.accepted_answer.accepter_username
+    if (this.siteSettings.show_who_marked_solved && this.args.post.topic.accepted_answer.accepter_username) {
```

Comment:
Should you have the same check of `this.siteSettings.display_name_on_posts` here as well, and use name instead in those cases? I hate how many name-related settings we have.. I'd think using `prioritize_name_in_ux` or whatever it is now would be better